src/test/groovy/graphql/language/AstPrinterTest.groovy

```diff
@@ -583,7 +625,17 @@ extend input Input @directive {
         String output = AstPrinter.printAstCompact(document)
 
         expect:
-        output == '''{aliasOfFoo:foo(arg1:"val1",args2:"val2") @isCached {hello} world @neverCache @okThenCache} fragment FX on SomeType {aliased:field(withArgs:"argVal",andMoreArgs:"andMoreVals")}'''
+        isParseableAst(output)
+        output == '''{aliasOfFoo:foo(arg1:"val1",args2:"val2") @isCached{hello}world @neverCache @okThenCache} fragment FX on SomeType {aliased:field(withArgs:"argVal",andMoreArgs:"andMoreVals")}'''
+    }
```

Comment:
You can also tighten directives

`{foo(arg1:"val1",args2:"val2") @isCached{hello}world @neverCache @okThenCache}}`

`{foo(arg1:"val1",args2:"val2")@isCached{hello}world@neverCache@okThenCache}}`